go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -185,6 +189,22 @@ func (bsnc *BaseSecondaryNetworkController) DeleteSecondaryNetworkResourceCommon
 			return err
 		}
 
+	case factory.IPAMClaimsType:
+		ipamClaim, ok := obj.(*ipamclaimsapi.IPAMClaim)
+		if !ok {
+			return fmt.Errorf("could not cast obj of type %T to *ipamclaimsapi.IPAMClaim", obj)
+		}
+		if err := bsnc.persistentIPsAllocator.Delete(ipamClaim); errors.Is(err, persistentips.ErrIgnoredIPAMClaim) {
+			klog.V(5).Infof(
+				"controller %q ignoring IPAMClaim for network: %s",
+				bsnc.NetInfo.GetNetworkName(),
+				ipamClaim.Spec.Network,
+			)
+		} else if err != nil {
+			return fmt.Errorf("error deleting IPAMClaim: %w", err)
+		}
+		return bsnc.persistentIPsAllocator.Delete(ipamClaim)
```

Comment:
We have delete it already

Reply:
? I don't follow. This returns the IP address to the IP pool.

Reply:
we are already calling ` bsnc.persistentIPsAllocator.Delete(ipamClaim)` bero the return.

Reply:
we should return nil here

Reply:
we're retuning nil below, after the switch case block.